source/portal/socketIOServer.js

```diff
@@ -355,6 +355,9 @@ var SocketIOServer = function(spec, portal, observer) {
   var startInsecure = function(port) {
     var server = require('http').createServer().listen(port);
     io = require('socket.io').listen(server, sioOptions);
+    io.origins((_, callback) => {
```

Comment:
My suggestion is that follow the example on socket.io document:
`io.origins((origin, callback) => {
  if origin in allowedList { // May use regExp to match the origin
    return callback(null, true);
  }
  callback('origin not allowed', false);
});`
https://socket.io/docs/v2/server-api/#server-origins-fn

Reply:
And add a configuration item in portal.toml